js/models/adaptModel.js

```diff
@@ -228,23 +228,17 @@ export default class AdaptModel extends LockingModel {
    */
   reset(type = 'hard', canReset = this.get('_canReset')) {
     if (!canReset) return false;
-    switch (type) {
-      case 'hard':
-      case true:
-        this.set({
-          _isEnabled: true,
-          _isComplete: false,
-          _isInteractionComplete: false
-        });
-        return true;
-      case 'soft':
-        this.set({
-          _isEnabled: true,
-          _isInteractionComplete: false
-        });
-        return true;
-    }
-    return false;
+    const resetData = {
+      _isEnabled: true,
+      _isInteractionComplete: false
+    };
+    const isHardReset = (type === 'hard' || type === true);
+    const isSoftReset = (type === 'soft');
+    if (!isHardReset && !isSoftReset) return false
```

Comment:
Minor: You can move the resetData object to after this early return. It isn't used otherwise